src/lib/hookify/hookedSelectors.ts

```diff
@@ -0,0 +1,229 @@
+import { useSyncExternalStore } from 'react';
+
+import { Action, Dispatch, EnhancedStore, UnknownAction } from '@reduxjs/toolkit';
+import { QueryClient, QueryObserver, useQuery } from '@tanstack/react-query';
+import { isFunction } from 'lodash';
+
+import { getUseBaseQuery } from './useBaseQuery';
+import hookifyHooks, { useSyncExternalStoreHf } from './vanillaHooks';
+
+type HookSub<Return> = (val: Return) => void;
+
+// wraps a function in vanilla hooks and returns an interface that lets users
+// set the arguments with call() and subscribe to all return values with subscribe()
+type Hookified<Return, Args extends any[]> = {
+  // callback whenever the return value changes for any reason
+  subscribe: (handle: HookSub<Return>) => () => void;
+
+  // will return undefined if never called or destroyed
+  getLatestValue: () => Return | undefined;
+
+  // sets the arguments and returns the first return value after running with these args
+  // further triggered changes available via subscribe()
+  call: (...args: Args) => Return;
+
+  // once torn down, cannot be restarted
+  tearDown: () => void;
+};
+
+function hookify<Return, Args extends any[]>(
+  hookFn: (...args: Args) => Return
+): Hookified<Return, Args> {
+  let destroyed = false;
+  let mostRecentResult: Return | undefined;
+  let subs: HookSub<Return>[] = [];
+  const removeSub = (subFn: HookSub<Return>) => {
+    subs = subs.filter((s) => s !== subFn);
+  };
+  const addSub = (sub: HookSub<Return>) => {
+    if (destroyed) throw new Error('This hookified fn is tore down');
+    subs.push(sub);
+    return () => removeSub(sub);
+  };
```

Comment:
super-nit: Rename `addSub` -> `subscribe` since it is re-assigned?

Reply:
wait if i'm reading this correctly (unless tyler just updated it) we're pushing to the subs array so it's not re-assigning